### PR TITLE
Machine Learning Managed Identity now correctly uses MSI's 2017 API instead of the 2019 API

### LINKED ISSUE #7597: MSI on Azure ML network unreachable
### Core Library

MSAL Node (@azure/msal-node)

### Core Library Version

3.2.3

### Wrapper Library

Not Applicable

### Wrapper Library Version

4.7.0

### Public or Confidential Client?

Confidential

### Description

We have recently upgraded our `@azure/identity` package from 4.3.0 to 4.7.0 which we use in Azure ML with a user assigned managed identity to fetch key vault credentials. The managed identity auth is now failing with network error. Reverting the version bump fixes the issue, but we cannot do this for other reasons.

### Error Message

_No response_

### MSAL Logs

azure:core-client:warning The baseUri option for SDK Clients has been deprecated, please use endpoint instead.
azure:core-client:warning The baseUri option for SDK Clients has been deprecated, please use endpoint instead.
azure:core-client:warning The baseUri option for SDK Clients has been deprecated, please use endpoint instead.
azure:core-rest-pipeline retryPolicy:info Retry 0: Attempting to send request 2ff1e8f6-fda2-42ab-b341-631e9c93487f
azure:keyvault-secrets:info Request: {
  "url": "https://vaultname.vault.azure.net/secrets/secretname/?api-version=7.5",
  "headers": {
    "accept": "application/json",
    "accept-encoding": "gzip,deflate",
    "user-agent": "azsdk-js-keyvault-secrets/4.8.0 core-rest-pipeline/1.10.1 Node/v20.18.1 OS/(x64-Linux-5.15.0-1079-azure)",
    "x-ms-client-request-id": "2ff1e8f6-fda2-42ab-b341-631e9c93487f"
  },
  "method": "GET",
  "timeout": 0,
  "disableKeepAlive": false,
  "streamResponseStatusCodes": {},
  "withCredentials": false,
  "tracingOptions": {
    "tracingContext": {
      "_contextMap": {}
    }
  },
  "requestId": "2ff1e8f6-fda2-42ab-b341-631e9c93487f",
  "allowInsecureConnection": false,
  "enableBrowserStreams": false
}
azure:core-rest-pipeline:info No cached TLS Agent exist, creating a new Agent
azure:keyvault-secrets:info Response status code: 401
azure:keyvault-secrets:info Headers: {
  "cache-control": "no-cache",
  "pragma": "no-cache",
  "content-type": "application/json; charset=utf-8",
  "expires": "-1",
  "x-ms-keyvault-region": "eastus",
  "x-ms-client-request-id": "2ff1e8f6-fda2-42ab-b341-631e9c93487f",
  "x-ms-request-id": "51bb9657-b216-4f80-87d6-92a9c3482bba",
  "x-ms-keyvault-service-version": "1.9.2103.1",
  "x-ms-keyvault-network-info": "conn_type=Ipv4;addr=20.3.30.51;act_addr_fam=InterNetwork;",
  "x-content-type-options": "nosniff",
  "strict-transport-security": "max-age=31536000;includeSubDomains",
  "www-authenticate": "Bearer authorization=\"https://login.microsoftonline.com/72f988bf-86f1-41af-91ab-2d7cd011db47\", resource=\"https://vault.azure.net\"",
  "date": "Wed, 26 Feb 2025 02:20:40 GMT",
  "content-length": "97"
}
azure:core-rest-pipeline retryPolicy:info Retry 0: Received a response from request 2ff1e8f6-fda2-42ab-b341-631e9c93487f
azure:core-rest-pipeline retryPolicy:info Retry 0: Processing 2 retry strategies.
azure:core-rest-pipeline retryPolicy:info Retry 0: Processing retry strategy throttlingRetryStrategy.
azure:core-rest-pipeline retryPolicy:info Retry 0: Skipped.
azure:core-rest-pipeline retryPolicy:info Retry 0: Processing retry strategy exponentialRetryStrategy.
azure:core-rest-pipeline retryPolicy:info Retry 0: Skipped.
azure:core-rest-pipeline retryPolicy:info None of the retry strategies could work with the received response. Returning it.
azure:identity:info ManagedIdentityCredential => getToken() => Using the MSAL provider for Managed Identity.
azure:identity:info ManagedIdentityCredential - Token Exchange => ManagedIdentityCredential - Token Exchange: Unavailable. The environment variables needed are: AZURE_CLIENT_ID (or the client ID sent through the parameters), AZURE_TENANT_ID and AZURE_FEDERATED_TOKEN_FILE
azure:identity:info ManagedIdentityCredential => getToken() => MSAL Identity source: MachineLearning
azure:identity:info ManagedIdentityCredential => getToken() => Calling into MSAL for managed identity token.
azure:identity:info ManagedIdentityCredential => MSAL Node V2 info message: [Wed, 26 Feb 2025 02:20:40 GMT] : [] : @azure/msal-node@3.2.3 : Info - [Managed Identity] ServiceFabric managed identity is unavailable because one or all of the 'IDENTITY_HEADER', 'IDENTITY_ENDPOINT' or 'IDENTITY_SERVER_THUMBPRINT' environment variables are not defined.
azure:identity:info ManagedIdentityCredential => MSAL Node V2 info message: [Wed, 26 Feb 2025 02:20:40 GMT] : [] : @azure/msal-node@3.2.3 : Info - [Managed Identity] AppService managed identity is unavailable because one or both of the 'IDENTITY_HEADER' and 'IDENTITY_ENDPOINT' environment variables are not defined.
azure:identity:info ManagedIdentityCredential => MSAL Node V2 info message: [Wed, 26 Feb 2025 02:20:40 GMT] : [] : @azure/msal-node@3.2.3 : Info - [Managed Identity] Environment variables validation passed for MachineLearning managed identity. Endpoint URI: http://localhost:46809/msi/token/. Creating MachineLearning managed identity.
azure:identity:info ManagedIdentityCredential => MSAL Node V2 info message: [Wed, 26 Feb 2025 02:20:40 GMT] : [] : @azure/msal-node@3.2.3 : Info - [Managed Identity] Adding user assigned client id to the request.
azure:core-rest-pipeline retryPolicy:info Retry 0: Attempting to send request 42e35aff-c26b-4c24-9998-c965011536dc
azure:core-rest-pipeline retryPolicy:info Retry 0: Attempting to send request 42e35aff-c26b-4c24-9998-c965011536dc
azure:core-rest-pipeline:info Request: {
  "url": "http://localhost:46809/MSI/token?api-version=2017-09-01&resource=REDACTED&client_id=REDACTED",
  "headers": {
    "metadata": "REDACTED",
    "secret": "REDACTED",
    "content-type": "application/x-www-form-urlencoded;charset=utf-8",
    "accept-encoding": "gzip,deflate",
    "user-agent": "azsdk-js-identity/4.7.0 core-rest-pipeline/1.10.1 Node/v20.18.1 OS/(x64-Linux-5.15.0-1079-azure)",
    "x-ms-client-request-id": "42e35aff-c26b-4c24-9998-c965011536dc"
  },
  "method": "GET",
  "timeout": 0,
  "disableKeepAlive": false,
  "withCredentials": false,
  "abortSignal": {},
  "requestId": "42e35aff-c26b-4c24-9998-c965011536dc",
  "allowInsecureConnection": true,
  "enableBrowserStreams": false
}
azure:core-rest-pipeline:info Response status code: 400
azure:core-rest-pipeline:info Headers: {
  "content-type": "text/plain; charset=utf-8",
  "date": "Wed, 26 Feb 2025 02:20:40 GMT",
  "content-length": "60"
}
azure:core-rest-pipeline retryPolicy:info Retry 0: Received a response from request 42e35aff-c26b-4c24-9998-c965011536dc
azure:core-rest-pipeline retryPolicy:info Retry 0: Processing 2 retry strategies.
azure:core-rest-pipeline retryPolicy:info Retry 0: Processing retry strategy throttlingRetryStrategy.
azure:core-rest-pipeline retryPolicy:info Retry 0: Skipped.
azure:core-rest-pipeline retryPolicy:info Retry 0: Processing retry strategy exponentialRetryStrategy.
azure:core-rest-pipeline retryPolicy:info Retry 0: Skipped.
azure:core-rest-pipeline retryPolicy:info None of the retry strategies could work with the received response. Returning it.
azure:core-rest-pipeline retryPolicy:info Retry 0: Received a response from request 42e35aff-c26b-4c24-9998-c965011536dc
azure:core-rest-pipeline retryPolicy:info Retry 0: Processing 1 retry strategies.
azure:core-rest-pipeline retryPolicy:info Retry 0: Processing retry strategy imdsRetryPolicy.
azure:core-rest-pipeline retryPolicy:info Retry 0: Skipped.
azure:core-rest-pipeline retryPolicy:info None of the retry strategies could work with the received response. Returning it.
azure:identity:error ManagedIdentityCredential => getToken() => ERROR. Scopes: https://vault.azure.net/.default. Error message: network_error: Network request failed.
azure:identity:info ChainedTokenCredential => getToken() => ERROR. Scopes: https://vault.azure.net/.default. Error message: ChainedTokenCredential authentication failed.
CredentialUnavailableError: ManagedIdentityCredential: Network unreachable. Message: network_error: Network request failed.
Error when setting up .env file: AggregateAuthenticationError: ChainedTokenCredential authentication failed.
CredentialUnavailableError: ManagedIdentityCredential: Network unreachable. Message: network_error: Network request failed
  errors: [
    CredentialUnavailableError: ManagedIdentityCredential: Network unreachable. Message: network_error: Network request failed
      [cause]: [ClientAuthError]
    }
  ]
}

### Network Trace (Preferrably Fiddler)

- [ ] Sent
- [ ] Pending

### MSAL Configuration

```javascript
MSAL is wrapped by @azure/identity package.
```

### Relevant Code Snippets

```javascript
if (process.env.AZURE_CLIENT_ID) {
		credentialOptions.push(new ManagedIdentityCredential({ clientId: process.env.AZURE_CLIENT_ID }));
}
```

### Reproduction Steps

1. Create Azure ML Compute Instance
2. Assign user assigned MI to Compute
3. Schedule job using @azure/identity package to use ManagedIdentityCredential

### Expected Behavior

@azure/identity can pick up MI to fetch secrets from KV

### Identity Provider

Entra ID (formerly Azure AD) / MSA

### Browsers Affected (Select all that apply)

None (Server)

### Regression

@azure/identity@4.3.0

### LINKED ISSUE #7627: Managed Identity: In the Machine Learning source, the client id is sent incorrectly in the network request
### Core Library

MSAL Node (@azure/msal-node)

### Core Library Version

latest

### Wrapper Library

Not Applicable

### Wrapper Library Version

N/A

### Public or Confidential Client?

Confidential

### Description

The Machine Learning source is based off of App Service 2017 (MSALs do not support this App Service 2017 version, only 2019). In the 2017 API, the client id should be passed to the network request as "clientid" instead of "client_id".

This bug was identified recently after Azure-SDK-for-JS removed their App Service 2017 implementation in favor msal-node's Machine Learning implementation. An Azure-SDK-for-JS customer is seeing network errors due to the incorrect way client id is being passed to the network request.

### Error Message

_No response_

### MSAL Logs

_No response_

### Network Trace (Preferrably Fiddler)

- [ ] Sent
- [ ] Pending

### MSAL Configuration

```javascript
N/A
```

### Relevant Code Snippets

```javascript
N/A
```

### Reproduction Steps

N/A

### Expected Behavior

N/A

### Identity Provider

Other

### Browsers Affected (Select all that apply)

None (Server)

### Regression

_No response_

### PR DESCRIPTION
**Pull request by Robbie-Microsoft**
Fixes #7597
Fixes #7627

### CODE REVIEW

File: lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts [hunk 1]
```diff
@@ -38,7 +38,8 @@ import { HttpClientWithRetries } from "../../network/HttpClientWithRetries.js";
  * Managed Identity User Assigned Id Query Parameter Names
  */
 export const ManagedIdentityUserAssignedIdQueryParameterNames = {
-    MANAGED_IDENTITY_CLIENT_ID: "client_id",
+    MANAGED_IDENTITY_CLIENT_ID_2017: "clientid",
+    MANAGED_IDENTITY_CLIENT_ID_2019: "client_id",
```
Comment (gladjohn) [inline lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts]:
```suggestion
    MANAGED_IDENTITY_CLIENT_ID: "client_id",
```

For app service 2017 or ML 2017, it makes sense to use the environment variable named MANAGED_IDENTITY_CLIENT_ID_2017. However, for other MSI sources (which may implement different versions of IMDS), using MANAGED_IDENTITY_CLIENT_ID_2019 might not be consistent or recognized. Therefore, sticking to a more generic, non-year-version-specific variable like MANAGED_IDENTITY_CLIENT_ID is usually the safest choice across different MSI providers.

File: lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts [hunk 2]
```diff
@@ -226,14 +227,17 @@ export abstract class BaseManagedIdentitySource {
...
     ): string {
         switch (managedIdentityIdType) {
             case ManagedIdentityIdType.USER_ASSIGNED_CLIENT_ID:
                 this.logger.info(
                     "[Managed Identity] Adding user assigned client id to the request."
                 );
-                return ManagedIdentityUserAssignedIdQueryParameterNames.MANAGED_IDENTITY_CLIENT_ID;
+                return api2017
+                    ? ManagedIdentityUserAssignedIdQueryParameterNames.MANAGED_IDENTITY_CLIENT_ID_2017
+                    : ManagedIdentityUserAssignedIdQueryParameterNames.MANAGED_IDENTITY_CLIENT_ID_2019;
```
Comment (gladjohn) [inline lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts]: same comment as above, you can just have this as `MANAGED_IDENTITY_CLIENT_ID`

File: lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts [hunk 3]
```diff
@@ -226,14 +227,17 @@ export abstract class BaseManagedIdentitySource {
 
     public getManagedIdentityUserAssignedIdQueryParameterKey(
         managedIdentityIdType: ManagedIdentityIdType,
-        imds?: boolean
+        imds?: boolean,
```
Comment (gladjohn) [inline lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts]: The parameter is named imds (true/false), but in practice this code handles multiple MSI sources (App Service, Azure Arc, etc.), not just IMDS. Because of that, imds can be misleading

File: lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts [hunk 4]
```diff
@@ -226,14 +227,17 @@ export abstract class BaseManagedIdentitySource {
...
+        api2017?: boolean
     ): string {
         switch (managedIdentityIdType) {
             case ManagedIdentityIdType.USER_ASSIGNED_CLIENT_ID:
                 this.logger.info(
                     "[Managed Identity] Adding user assigned client id to the request."
                 );
-                return ManagedIdentityUserAssignedIdQueryParameterNames.MANAGED_IDENTITY_CLIENT_ID;
+                return api2017
+                    ? ManagedIdentityUserAssignedIdQueryParameterNames.MANAGED_IDENTITY_CLIENT_ID_2017
```
Comment (gladjohn) [inline lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts:243]: add a comment to which MSI sources require the clientid (2017) parameter vs. client_id and mention that App Service, Azure Arc, and others share the client_id style. This helps future maintainers quickly see why the code is split.

File: lib/msal-node/test/client/ManagedIdentitySources/AppService.spec.ts [hunk 5]
```diff
@@ -72,6 +77,15 @@ describe("Acquires a token successfully via an App Service Managed Identity", ()
             expect(networkManagedIdentityResult.accessToken).toEqual(
                 DEFAULT_USER_SYSTEM_ASSIGNED_MANAGED_IDENTITY_AUTHENTICATION_RESULT.accessToken
             );
+
+            const url: URLSearchParams = new URLSearchParams(
+                sendGetRequestAsyncSpy.mock.lastCall[0]
+            );
+            expect(
+                url.has(
+                    ManagedIdentityUserAssignedIdQueryParameterNames.MANAGED_IDENTITY_CLIENT_ID_2019
```
Comment (gladjohn) [inline lib/msal-node/test/client/ManagedIdentitySources/AppService.spec.ts]: even in tests you can use MANAGED_IDENTITY_CLIENT_ID

File: lib/msal-node/test/client/ManagedIdentitySources/MachineLearning.spec.ts [hunk 6]
```diff
@@ -71,6 +76,15 @@ describe("Acquires a token successfully via an Machine Learning Managed Identity
...
             );
+
+            const url: URLSearchParams = new URLSearchParams(
+                sendGetRequestAsyncSpy.mock.lastCall[0]
+            );
+            expect(
+                url.has(
+                    ManagedIdentityUserAssignedIdQueryParameterNames.MANAGED_IDENTITY_CLIENT_ID_2017
+                )
+            ).toBe(true);
```
Comment (gladjohn) [inline lib/msal-node/test/client/ManagedIdentitySources/MachineLearning.spec.ts:87]: While the test confirms the “happy path” for ML 2017, it doesn’t confirm that the code avoided incorrectly including “client_id” at the same time. Would it be good to check for,  

```js
expect(
  url.has(
    ManagedIdentityUserAssignedIdQueryParameterNames.MANAGED_IDENTITY_CLIENT_ID
  )
).toBe(false);

```

File: lib/msal-node/test/client/ManagedIdentitySources/ServiceFabric.spec.ts [hunk 7]
```diff
@@ -78,6 +83,15 @@ describe("Acquires a token successfully via an App Service Managed Identity", ()
             expect(networkManagedIdentityResult.accessToken).toEqual(
                 DEFAULT_USER_SYSTEM_ASSIGNED_MANAGED_IDENTITY_AUTHENTICATION_RESULT.accessToken
             );
+
+            const url: URLSearchParams = new URLSearchParams(
+                sendGetRequestAsyncSpy.mock.lastCall[0]
+            );
+            expect(
+                url.has(
+                    ManagedIdentityUserAssignedIdQueryParameterNames.MANAGED_IDENTITY_CLIENT_ID_2019
```
Comment (gladjohn) [inline lib/msal-node/test/client/ManagedIdentitySources/ServiceFabric.spec.ts]: same comment as earlier test, can we check, 

```js
expect(
  url.has(
    ManagedIdentityUserAssignedIdQueryParameterNames.MANAGED_IDENTITY_CLIENT_ID_2017
  )
).toBe(false);

```

Comment (gladjohn) [reply lib/msal-node/test/client/ManagedIdentitySources/ServiceFabric.spec.ts]: checking this for all other non ML 2017 sources

**Comment by gladjohn** [comment]
Overall, the PR looks good. The main suggestions are to add clarity for future maintainers and tighten test coverage with a negative assertion (if needed) and to fix any variable names that may be misleading with versioning in them.

File: lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts (hunk 3 above)
Comment (gladjohn) [reply lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts:230]: also `credential` source will be 2023 version I think

File: lib/msal-node/test/client/ManagedIdentitySources/MachineLearning.spec.ts (hunk 6 above)
Comment (gladjohn) [reply lib/msal-node/test/client/ManagedIdentitySources/MachineLearning.spec.ts:87]: you can see how in MSAL .NET we test this. i.e. testing the not expected param value - https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/5fcc057c08fbb8cf4fe7a231558557be398dddaa/tests/Microsoft.Identity.Test.Common/Core/Mocks/MockHttpMessageHandler.cs#L125

File: lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts (hunk 1 above)
Comment (gladjohn) [reply lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts]: also credential source will be 2023 version I think

File: lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts [hunk 8]
```diff
@@ -226,14 +227,17 @@ export abstract class BaseManagedIdentitySource {
...
     public getManagedIdentityUserAssignedIdQueryParameterKey(
         managedIdentityIdType: ManagedIdentityIdType,
-        imds?: boolean
+        imds?: boolean,
+        api2017?: boolean
     ): string {
         switch (managedIdentityIdType) {
             case ManagedIdentityIdType.USER_ASSIGNED_CLIENT_ID:
                 this.logger.info(
                     "[Managed Identity] Adding user assigned client id to the request."
```
Comment (gladjohn) [inline lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts]: can we make the logging different to identify which `USER_ASSIGNED_CLIENT_ID` was added? 2017 vs newer?

File: lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts [hunk 9]
```diff
@@ -38,7 +38,8 @@ import { HttpClientWithRetries } from "../../network/HttpClientWithRetries.js";
  * Managed Identity User Assigned Id Query Parameter Names
  */
 export const ManagedIdentityUserAssignedIdQueryParameterNames = {
-    MANAGED_IDENTITY_CLIENT_ID: "client_id",
+    MANAGED_IDENTITY_CLIENT_ID_2017: "clientid",
```
Comment (gladjohn) [inline lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts]: also adding a comment here to explain why we have these two might help in the future.

**Comment by Robbie-Microsoft** [comment]
This has been blocked for over a week. Customer is not being responsive, and I can't get instructions on how to set up a ML cluster to manually test this.

File: lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts (hunk 1 above)
Comment (Robbie-Microsoft) [reply lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts]: done

File: lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts (hunk 3 above)
Comment (Robbie-Microsoft) [reply lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts]: done

File: lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts (hunk 4 above)
Comment (Robbie-Microsoft) [reply lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts:243]: done

File: lib/msal-node/test/client/ManagedIdentitySources/AppService.spec.ts (hunk 5 above)
Comment (Robbie-Microsoft) [reply lib/msal-node/test/client/ManagedIdentitySources/AppService.spec.ts]: done

File: lib/msal-node/test/client/ManagedIdentitySources/MachineLearning.spec.ts (hunk 6 above)
Comment (Robbie-Microsoft) [reply lib/msal-node/test/client/ManagedIdentitySources/MachineLearning.spec.ts:87]: done

File: lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts (hunk 8 above)
Comment (Robbie-Microsoft) [reply lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts]: done

File: lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts (hunk 9 above)
Comment (Robbie-Microsoft) [reply lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts]: done

File: lib/msal-node/test/client/ManagedIdentitySources/ServiceFabric.spec.ts (hunk 7 above)
Comment (Robbie-Microsoft) [reply lib/msal-node/test/client/ManagedIdentitySources/ServiceFabric.spec.ts]: done

File: lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts (hunk 2 above)
Comment (Robbie-Microsoft) [reply lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts]: done

**Comment by gladjohn** [comment]
```
Fixes https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/7597
Fixes https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/7627
```
we should close one of them as dupe

**Review by gladjohn — APPROVED** [review]
looks good @Robbie-Microsoft 

The latest changes address all earlier feedback, including separating out the clientid vs. client_id usage, improving logs, and adding negative tests. 

As a minor suggestion, you might consider logging the explicit API version strings ("2017-09-01" vs. "2019-08-01") to further clarify debugging if not done already, but that’s optional. Overall, this is ready to merge. Good work!

**Comment by Robbie-Microsoft** [comment]
> ```
> Fixes https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/7597
> Fixes https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/7627
> ```
> 
> we should close one of them as dupe

I'm not going to mark them as duplicate because this PR might not fix
